src/Catalyst.Core.Lib/Util/Duration.cs

```diff
@@ -0,0 +1,201 @@
+        ///   <para>
+        ///   A duration string is a possibly signed sequence of decimal numbers, 
+        ///   each with optional fraction and a unit suffix, such as "300ms", "-1.5h" or "2h45m". 
+        ///   Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+        ///   </para>
+        /// </remarks>
+        public static TimeSpan Parse(string s)
+        {
+            if (string.IsNullOrWhiteSpace(s) || s == "n/a" || s == "unknown")
+                return TimeSpan.Zero;
+
+            var result = TimeSpan.Zero;
+            var negative = false;
+            using (var sr = new StringReader(s))
+            {
+                if (sr.Peek() == '-')
+                {
+                    negative = true;
+                    sr.Read();
+                }
+
+                while (sr.Peek() != -1)
+                {
+                    result += ParseComponent(sr);
+                }
+            }
+
+            if (negative)
+                return TimeSpan.FromTicks(-result.Ticks);
+
+            return result;
+        }
+
+        private static TimeSpan ParseComponent(StringReader reader)
+        {
+            var value = ParseNumber(reader);
+            var unit = ParseUnit(reader);
+
+            switch (unit)
+            {
+                case "h":
+                    return TimeSpan.FromHours(value);
+                case "m":
+                    return TimeSpan.FromMinutes(value);
+                case "s":
+                    return TimeSpan.FromSeconds(value);
+                case "ms":
+                    return TimeSpan.FromMilliseconds(value);
+                case "us":
+                case "µs":
+                    return TimeSpan.FromTicks((long) (value * TicksPerMicrosecond));
+                case "ns":
+                    return TimeSpan.FromTicks((long) (value * TicksPerNanosecond));
+                case "":
+                    throw new FormatException("Missing IPFS duration unit.");
+                default:
+                    throw new FormatException($"Unknown IPFS duration unit '{unit}'.");
+            }
+        }
+
+        private static double ParseNumber(StringReader reader)
+        {
+            var s = new StringBuilder();
+            while (true)
+            {
+                var c = (char) reader.Peek();
+                if (char.IsDigit(c) || c == '.')
+                {
+                    s.Append(c);
+                    reader.Read();
+                    continue;
+                }
+
+                return double.Parse(s.ToString(), CultureInfo.InvariantCulture);
+            }
+        }
+
+        private static string ParseUnit(StringReader reader)
+        {
+            var s = new StringBuilder();
+            while (true)
+            {
+                var c = (char) reader.Peek();
+                if (char.IsDigit(c) || c == '.' || c == (char) 0xFFFF)
+                    break;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor the code in order to remove this break statement.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)